### DISCUSSION TITLE
Stop processing after first occurence

### DISCUSSION THREAD

**Discussion by ottobaer** [jaqx0r/mtail#114] [Q&A] [answered]
I'm trying to find a solution to stop processing after the first occurence of a match. 

I have a log file where the same expression is repeated on multiple lines and so the same information gets put into the same metric multiple times. 

I tried using `stop` after the regex, but that does nothing it seems. 

Any ideas?

**Comment by jaqx0r** [comment, marked as answer]
That's not what `mtail` is for.  It is for continuously scanning log lines to extract internal state from an otherwise opaque process.

`stop` ends the program when executing on the current line, it doesn't stop `mtail` from reading new lines in a log stream.

Why do you want to stop processing after the first occurrence of a match?

**Comment by jaqx0r** [comment]
I never got a response so this discussion is closed.
